BuildingAndTestingInBunsen.md

```diff
@@ -0,0 +1,119 @@
+# Building and Testing the VDO software in a Bunsen Environment
+
+This guide is intended to help get you started with developing, building, and testing your contributions to VDO in a Bunsen environment.
+
+# Prerequisites
+
+It is expected that you have satisfied the following criteria before getting started:
+
+* [Set up a Bunsen Environment](Getting_Started.md)
+
+# Expectations of this Guide
+
+This guide is currently intended to provide enough information to get minimally functional.  A more detailed guide will be generated later, but in the meantime this document should help you get started if you're interested in contributing to the VDO projects.
+
+# Find the appropriate code repository
+
+Identify which repository you want to test.
+
+Currently the following VDO repositories are set up to allow for local testing:
+* [vdo-devel](https://github.com/dm-vdo/vdo-devel) - This is where the VDO product code resides as well as any supporting test code that is only needed for VDO itself.
+* [common](https://github.com/dm-vdo/common) - This is where the library code resides that supports both the test environment as well as VDO.
+
+* How do I tell what's the difference between the two?
+  * The first thing to figure out is whether you're modifying the VDO code itself or the tests that directly test VDO.
+    * If that's the case, then you want to use the [vdo-devel](https://github.com/dm-vdo/vdo-devel) tree.
+  * Otherwise, you'll need to look for the code in either repository.  If you're not sure, then you should likely start on the [vdo-devel](https://github.com/dm-vdo/vdo-devel) repository and move to the [common](https://github.com/dm-vdo/common) repository if you still couldn't find what you were looking for.
+
+# Find the code
+
+For the purposes of this guide, all of the work you will be doing is going to take place on the Bunsen `resource` system.
+
+SSH to the `resource` system (typically named `ossbunsen-resource`) as the user that you have provisioned it with.
+
+
+For changes to the [vdo-devel](https://github.com/dm-vdo/vdo-devel) repository:
+
+  * Starting in the home directory, clone the repository and cd into it
+    ```
+    git clone https://github.com/dm-vdo/vdo-devel.git && cd vdo-devel
+    ```
+  * The files under src from here can be generally distinguished by language-type (though `c++` isn't accurate).
+    * Core VDO code can be found under src/c++/vdo
+    * Perl test code can be found under src/perl/Permabit
```

Comment:
Suggest: "Perl test library code can be found under src/perl/Permabit. Core perl test code can be found under src/perl/vdotest".

Reply:
Yeah, that's a reasonable way to phrase it.  I'll include that in an update momentarily.  Thanks for the suggestion.

Reply:
Should be in the latest update contained within this PR.